actors/miner/tests/compact_sector_numbers_tests.rs

```diff
@@ -0,0 +1,155 @@
+use fil_actors_runtime::test_utils::{expect_abort, MockRuntime};
+use fvm_shared::address::Address;
+use fvm_shared::sector::MAX_SECTOR_NUMBER;
+use fvm_shared::{clock::ChainEpoch, econ::TokenAmount, error::ExitCode};
+
+mod util;
+use util::*;
+
+const PERIOD_OFFSET: ChainEpoch = 100;
+
+fn setup() -> (ActorHarness, MockRuntime) {
+    let big_balance = 20u128.pow(23);
+
+    let h = ActorHarness::new(PERIOD_OFFSET);
+    let mut rt = h.new_runtime();
+    h.construct_and_verify(&mut rt);
+    rt.balance.replace(TokenAmount::from(big_balance));
+
+    (h, rt)
+}
+
+mod compact_sector_numbers_test {
+    use super::*;
+
+    #[test]
+    fn compact_sector_numbers_then_pre_commit() {
+        // Create a sector.
+        let (mut h, mut rt) = setup();
+        let all_sectors =
+            h.commit_and_prove_sectors(&mut rt, 1, DEFAULT_SECTOR_EXPIRATION, vec![], true);
+
+        let target_sector_num = all_sectors[0].sector_number;
+        h.compact_sector_numbers(
+            &mut rt,
+            h.worker,
+            bitfield_from_slice(&[target_sector_num, target_sector_num + 1]),
+        );
+
+        let precommit_epoch = rt.epoch;
+        let deadline = h.deadline(&rt);
+        let expiration = deadline.period_end()
+            + DEFAULT_SECTOR_EXPIRATION as i64 * rt.policy.wpost_proving_period;
+
+        // Allocating masked sector number should fail.
+        {
+            let precommit = h.make_pre_commit_params(
+                target_sector_num,
```

Comment:
This needs to be target_sector_num + 1, target_sector_num is already taken up by the first commit so failing would happen without the compact sector numbers call.  This test wants to make sure that the newly masked out sector number though unassigned to an actual sector still fails.